barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_common.hpp

```diff
@@ -15,9 +15,8 @@ namespace bb::avm_trace {
 using Flavor = bb::AvmFlavor;
 using FF = Flavor::FF;
 
-} // namespace bb::avm_trace
-
-namespace bb::avm_trace {
+// To toggle all relevant unit tests with proving, set the env variable "AVM_TESTS_ENABLE_PROVING".
+static const bool ENABLE_PROVING = std::getenv("AVM_TESTS_ENABLE_PROVING") != nullptr;
```

Comment:
Could we call this one AVM_ENABLE_FULL_PROVING?

It both (1) enables proving in tests, and (2) enables full-sized trace in both tests and non-avm-cpp-test code (e.g., e2e tests, production, etc)

Reply:
Also since proving a function only takes 1m, maybe enable it here as well?

https://github.com/AztecProtocol/aztec-packages/blob/d4921a032a56eb0ba464b0e505f6ac11cb41502d/yarn-project/end-to-end/src/e2e_prover/full.test.ts#L9